misc/server/CHANGELOG.md

```diff
@@ -16,14 +11,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 [PR 4467]: https://github.com/libp2p/rust-libp2p/pull/4467
 
-## [0.12.2]
+## 0.12.2
```

Comment:
Long term I suggest we switch to https://keepachangelog.com/en/1.0.0/

Short term, thanks for the consistency.